### PR TITLE
PTM: Patch for the panic in math operations

### PR DESCRIPTION
**Pull request by satyamakgec**

### CODE REVIEW

File: contracts/PercentageTransferManager/Cargo.toml [hunk 1]
```diff
@@ -38,6 +38,10 @@ std = [
     "scale-info",
     "scale-info/std",
 ]
+
+[profile.release]
+overflow-checks = false
```
Comment (Centril) [inline contracts/PercentageTransferManager/Cargo.toml:43]: I can see how wrapping is right in mainnet.
But are we OK with silent wrapping arithmetic in a testnet?
Panicking might help us catch bugs in the interim.

File: contracts/PercentageTransferManager/src/lib.rs [hunk 2]
```diff
@@ -139,7 +139,7 @@ mod percentage_transfer_manager {
             _current_holder_count: u64,
         ) -> RestrictionResult {
             if from == None && self.allow_primary_issuance
-                || self.is_exempted_or_not(&(to.unwrap_or_default()))
+                || to.map_or(false, |to| self.is_exempted_or_not(&to))
```
Comment (Centril) [inline contracts/PercentageTransferManager/src/lib.rs:142]:
```suggestion
                || to.map_or(false, |to| self.is_exempted(&to))
```

File: contracts/PercentageTransferManager/Cargo.toml (hunk 1 above)
Comment (adamdossa) [reply contracts/PercentageTransferManager/Cargo.toml:43]: Can you ELI5 this flag please?

Comment (satyamakgec) [reply contracts/PercentageTransferManager/Cargo.toml:43]: This flag is introduced to overcome the rustc bug related to wasm. 

`overflow-checks` are checks inserted by the Rust compiler for integer arithmetic. those do panic. if there was no bug we would have direct aborts instead of panics in our Wasm builds, but due to this bug not all of those get filtered out leaving us with those strange errors {Simillar Remon faced today} not inserting those overflow checks in the first place simply removes some category of potential places where the Rust compiler can fail.
in Solidity speak: overflow-checks is similar to safe-math`

Comment (Centril) [reply contracts/PercentageTransferManager/Cargo.toml:43]: If I remember correctly, with the flag set to true, arithmetic over or underflow will result in panics. When set to false, it will instead result in wrapping arithmetic. So `0 - 1 == usize::MAX` for example

Comment (maxsam4) [reply contracts/PercentageTransferManager/Cargo.toml:43]: Why was an overflow being triggered in the first place? I don't think disabling overflow checks is the right solution.

Comment (satyamakgec) [reply contracts/PercentageTransferManager/Cargo.toml:43]: So there is a statement in the contract i.e `10u128.pow(6)` which is a valid statement if we compile it into rust but during wasm compilation, it panics and colludes the wasm blob. It is a bug in the compiler so make it work I switch off the `overflow-checks` but another way we can achieve this by directly using `1000000` directly without using the `pow` math operation.

Comment (maxsam4) [reply contracts/PercentageTransferManager/Cargo.toml:43]: Let's directly use `1_000_000` and report this upstream.

Comment (satyamakgec) [reply contracts/PercentageTransferManager/Cargo.toml:43]: It is already a well-known issue and Robin suggested to use `overflow-checks`

Comment (satyamakgec) [reply contracts/PercentageTransferManager/Cargo.toml:43]: But anyway I am fine with `1_000_000` this also

Comment (satyamakgec) [reply contracts/PercentageTransferManager/Cargo.toml:43]: Hmm.. This will also not work as well I thought it will work but it isn't. Somehow there is a problem in the arithmetic operation to make it compile in wasm

Comment (maxsam4) [reply contracts/PercentageTransferManager/Cargo.toml:43]: Then this pow thing wasn't the issue. It being a known problem does not make it safe. We need to find the actual loc that is causing this issue.

**Review by maxsam4 — CHANGES_REQUESTED** [review]
Let's try to figure out the actual cause of panic or think of something else.

**Review by adamdossa — APPROVED** [review]

**Review by maxsam4 — APPROVED** [review]
As per the convo, approving as a temporary fix, need to revisit later.
